pool/swap_test.gno

```diff
@@ -0,0 +1,1079 @@
+			token1: u256.Zero(),
+		},
+		feeGrowthGlobal0X128: u256.Zero(),
+		feeGrowthGlobal1X128: u256.Zero(),
+		tickBitmaps:          make(TickBitmaps),
+		ticks:                make(Ticks),
+		positions:            make(Positions),
+	}
+
+	originalGetPool := GetPool
+	defer func() {
+		GetPool = originalGetPool
+	}()
+	GetPool = func(token0Path, token1Path string, fee uint32) *Pool {
+		return mockPool
+	}
+
+	tests := []struct {
+		name              string
+		token0Path        string
+		token1Path        string
+		fee               uint32
+		zeroForOne        bool
+		amountSpecified   string
+		sqrtPriceLimitX96 string
+		expectAmount0     string
+		expectAmount1     string
+		expectSuccess     bool
+	}{
+		{
+			name:              "zero amount token0 to token1",
+			token0Path:        "token0",
+			token1Path:        "token1",
+			fee:               3000,
+			zeroForOne:        true,
+			amountSpecified:   "0",
+			sqrtPriceLimitX96: "900000000000000000",
+			expectAmount0:     "0",
+			expectAmount1:     "0",
+			expectSuccess:     false,
+		},
+		{
+			name:              "insufficient balance",
+			token0Path:        "token0",
+			token1Path:        "token1",
+			fee:               3000,
+			zeroForOne:        false,
+			amountSpecified:   "2000000000",
+			sqrtPriceLimitX96: "1100000000000000000",
+			expectAmount0:     "0",
+			expectAmount1:     "0",
+			expectSuccess:     false,
+		},
+		{
+			name:              "insufficient balance token1 to token0",
+			token0Path:        "token0",
+			token1Path:        "token1",
+			fee:               3000,
+			zeroForOne:        false,
+			amountSpecified:   "3000000000",
+			sqrtPriceLimitX96: "1100000000000000000",
+			expectAmount0:     "0",
+			expectAmount1:     "0",
+			expectSuccess:     false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			amount0, amount1, success := DrySwap(
+				tt.token0Path,
+				tt.token1Path,
+				tt.fee,
+				tt.zeroForOne,
+				tt.amountSpecified,
+				tt.sqrtPriceLimitX96,
+			)
+
+			uassert.Equal(t, success, tt.expectSuccess)
+			uassert.Equal(t, amount0, tt.expectAmount0)
+			uassert.Equal(t, amount1, tt.expectAmount1)
+		})
+	}
+}
+
+func TestSwapAndDrySwapComparison(t *testing.T) {
+	const addr = pusers.AddressOrName(consts.ROUTER_ADDR)
+
+	tests := []struct {
+		name              string
+		setupFn           func(t *testing.T)
+		token0Path        string
+		token1Path        string
+		fee               uint32
+		zeroForOne        bool
+		amountSpecified   string
+		sqrtPriceLimitX96 string
+	}{
+		{
+			name: "normal swap",
+			setupFn: func(t *testing.T) {
+				InitialisePoolTest(t)
+				TokenFaucet(t, gnsPath, addr)
+				TokenApprove(t, gnsPath, addr, pool, uint64(1000))
+			},
+			token0Path:        wugnotPath,
+			token1Path:        gnsPath,
+			fee:               fee3000,
+			zeroForOne:        false,
+			amountSpecified:   "100",
+			sqrtPriceLimitX96: "1461446703485210103287273052203988822378723970341",
```

Comment:
seems like max price for zeroForOne true, add comments or make it to variable for better readability

Reply:
@r3v4s https://github.com/gnoswap-labs/gnoswap/pull/421/commits/906ab514c514565999bc48dc139a8a847b3e4422